src/sp_disabled_functions.c

```diff
@@ -501,7 +501,7 @@ int hook_disabled_functions(void) {
   ret |= hook_functions(SNUFFLEUPAGUS_G(config)
                             .config_disabled_functions_ret->disabled_functions);
 
-  if (NULL != SNUFFLEUPAGUS_G(config).config_eval->blacklist->data) {
+  if (NULL != SNUFFLEUPAGUS_G(config).config_eval->blacklist && NULL != SNUFFLEUPAGUS_G(config).config_eval->blacklist->data) {
```

Comment:
Why a second condition?

Reply:
Because even if blacklist is not NULL you still want to check for blacklist->data. If you remove it, you change the behavior.

Reply:
In what case could I have a non-NULL list with no data in it?